### DISCUSSION TITLE
Thoughts on react 19 actions?

### DISCUSSION THREAD

**Discussion by jmatsushita** [pmndrs/jotai#2525] [Q&A]
The [recent React 19 beta announcement](https://react.dev/blog/2024/04/25/react-19#actions) proposes Actions as "functions that use async transitions". What are your thoughts regarding how this will work with Jotai?

**Comment by dai-shi** [comment]
Jotai has been developed for transitions (Concurrent React / Concurrent Mode) from day 1.
It doesn't have loading state or error state. (unless you use `loadable` util.)
It works with `useTransition`, and probably with `useOptimistic` (I've never tried.)

There might be a new pattern for using `startTransition` internally, and a better support 
for `action={}` prop, but we don't know best practice yet.
